### PR TITLE
Migrate from NextJS to ReactJS

### PR DESCRIPTION
**Pull request by MySmilingTurban**
This PR contains commits which change the codebase from NextJS to ReactJS

Expected behaviour:
- one-page web app with "Welcome to Shabadavali" text to test deployment (PR deploys shows empty site, wanted to check merge deploy)
- contains configuration for CI/CD pipeline, eslint, Tailwind CSS, and rendering of SCSS files with React

### CODE REVIEW

**Comment by lbk003** [comment]
@MySmilingTurban - can you please capture the problems in the PR w/ the firebase deploy ?

**Comment by MySmilingTurban** [comment]
> @MySmilingTurban - can you please capture the problems in the PR w/ the firebase deploy ?

As we can see, the action works smoothly on pull request and it also gives us a preview channel URL. Unfortunately, if we navigate to that link, it shows Firebase site not found.

I tried deploying the same to the default site, i.e., gurmukhi-dev, and it works fine.
I am looking into documentations as it seems like an internal issue because ideally if I am specifying the target site for deployment, it should be there on deployment.

File: firebase.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "firestore": {
+    "rules": "firestore.rules",
+    "indexes": "firestore.indexes.json"
+  },
+  "hosting": {
+    "target": "default",
```
[the remark this answers is not in the document]
Comment (MySmilingTurban) [reply firebase.json:7]: our problem mostly revolves around [this](https://firebase.google.com/docs/hosting/multisites#set_up_deploy_targets)

File: .github/workflows/firebase-hosting-merge.yml
Comment (lbk003) [inline .github/workflows/firebase-hosting-merge.yml:1]: Remove this from this PR. Lets get hosting working first

File: .github/workflows/firebase-hosting-pull-request.yml
Comment (lbk003) [inline .github/workflows/firebase-hosting-pull-request.yml:1]: Also, remove this file. We will add it back later once we have deploy working

**Review by lbk003 — CHANGES_REQUESTED** [review]

File: firestore.rules [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      // This rule allows anyone with your database reference to view, edit,
+      // and delete all data in your database. It is useful for getting
+      // started, but it is configured to expire after 30 days because it
+      // leaves your app open to attackers. At that time, all client
+      // requests to your database will be denied.
+      //
+      // Make sure to write security rules for your app before that time, or
+      // else all client requests to your database will be denied until you
+      // update your rules.
+      allow read, write: if request.auth != null;
```
[the remark this answers is not in the document]
Comment (MySmilingTurban) [reply firestore.rules:14]: Firestore is working fine

File: src/constants/wordsData.ts [hunk 3]
```diff
@@ -46,12 +46,12 @@ export const wordData: WordData = {
       },
       {
         sentence: 'ਮੈਂ ਤੇ ਗੁਰਵਿੰਦਰ ਕੁਸ਼ਤੀ ਦੇ ਅਭਿਆਸ ਲਈ ਹਰ ਰੋਜ਼ ਅਖਾੜੇ ਜਾਂਦੇ ਹਾਂ।',
-        sentenceEnglish: 'Gurvinder and I go to the ring every day to practice wrestling.'
+        sentenceEnglish: 'Gurvinder and I go to the ring every day to practice wrestling.',
```
[the remark this answers is not in the document]
Comment (MySmilingTurban) [reply src/constants/wordsData.ts:49]: Hnji, will be adding it after static web pages setup

File: firebase.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (amitojsingh) [reply firebase.json:7]: @MySmilingTurban  is it fixed veer ji?

File: public/404.html [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>Page Not Found</title>
+
+    <style media="screen">
+      body { background: #ECEFF1; color: rgba(0,0,0,0.87); font-family: Roboto, Helvetica, Arial, sans-serif; margin: 0; padding: 0; }
```
Comment (amitojsingh) [inline public/404.html:9]: Veer ji can we use an external stylesheet here?

File: tsconfig.json [hunk 5]
```diff
@@ -26,10 +29,10 @@
...
       ]
-    }
+    },
   },
   "include": [
-    "next-env.d.ts",
+    "react-app-env.d.ts",
     "**/*.ts",
     "**/*.tsx",
     ".next/types/**/*.ts",
```
Comment (amitojsingh) [inline tsconfig.json:38]: Veer ji, I belive we don't need this line.

**Review by amitojsingh — CHANGES_REQUESTED** [review]

**Review by amitojsingh — APPROVED** [review]

**Pull request merged**